### DISCUSSION TITLE
Creating more e2e and integrations tests

### DISCUSSION THREAD

**Discussion by Shashwat79802** [nocodb/nocodb#6992] [General]
I browsed through the repository and noticed that are are very few e2e tests and I could find any integration tests. Can I contribute to the project by adding these tests? I use a tool called keploy for testing and generating data mocks and it'd be more helpful to have these tests in the project.

**Comment by o1lab** [comment]
@Shashwat79802 : You have created same discussion thread in many repos in the pretext to introduce keploy with whom you are employed. Did you really take a look into our tests and what is our test stack etc ? We've quite a few e2e tests and API tests. Introducing a new test framework requires significant effort to introduce & maintain them. So we are happy to take PRs with existing frameworks.

<img width="804" alt="Screenshot 2023-11-16 at 08 40 44" src="https://github.com/nocodb/nocodb/assets/5435402/12445d70-5fa0-49cd-aecf-1e0d09f6cee2">

**Reply by Shashwat79802** [reply]
I appreciate your concerns about introducing a new test framework and the importance of maintaining existing practices. Could we create a dedicated branch to experiment with a few Keploy tests? If they align with your standards and provide you with better coverage than earlier, then we can discuss further. This way, we can assess the impact without a full commitment upfront.
